app/src/main/java/woowacourse/shopping/ui/vmfactory/AutoViewModelFactory.kt

```
@@ -0,0 +1,32 @@
+        val constructor = viewModel.primaryConstructor
+            ?: viewModel.constructors.firstOrNull()
+            ?: error("${viewModel.simpleName} 생성자를 찾을 수 없습니다")
+
+        val constructorArguments: Array<Any?> =
+            constructor.parameters.map { parameter: KParameter ->
+                val requiredType = parameter.type.classifier as? KClass<*>
+                    ?: error("${parameter.type} 지원하지 않는 파라미터 입니다.")
+
+                try {
+                    container.get(requiredType)
+                } catch (e: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)